app/controllers/middleware_server_controller.rb

```diff
@@ -74,12 +73,12 @@ def show
 
     if @display == 'middleware_datasources'
       @gtl_url = '/show'
-      show_container_display(@record, 'middleware_datasource', MiddlewareDatasource)
+      show_middleware_display(@record, 'middleware_datasource', MiddlewareDatasource)
```

Comment:
this and line 79 are very similar except for entity names. so you could go for a switch case, unify them and then generically send parameters to `show_middleware_display` (for example, instead of hardcoding middleware_datasource string, you could do something around calling `downcase`, etc.)